packages/runtime/src/lib/asa.ts

```diff
@@ -14,57 +14,81 @@ const ASSETS_DIR = "assets";
  * @param filename asa filename
- * @param asaDef asset definitions
+ * @param asaDef asset definition
  */
-function validateSingle (accounts: AccountMap | RuntimeAccountMap, filename: string, asaDef: ASADef): void {
+function validateOptInAccNames (accounts: AccountMap | RuntimeAccountMap,
+  asaDef: ASADef,
+  source?: string): void {
   if (!asaDef.optInAccNames || asaDef.optInAccNames.length === 0) {
     return;
   }
   for (const accName of asaDef.optInAccNames) {
     if (!accounts.get(accName)) {
       throw new RuntimeError(
         RUNTIME_ERRORS.ASA.PARAM_ERROR_NO_NAMED_OPT_IN_ACCOUNT, {
-          filename: filename,
+          source: source,
           optInAccName: accName
         });
     }
   }
 }
 
-function validateParsedASADefs (
-  accounts: AccountMap | RuntimeAccountMap, asaDefs: ASADefs, filename: string): void {
-  for (const def of Object.values(asaDefs)) {
-    def.manager = def.manager !== "" ? def.manager : undefined;
-    def.reserve = def.reserve !== "" ? def.reserve : undefined;
-    def.freeze = def.freeze !== "" ? def.freeze : undefined;
-    def.clawback = def.clawback !== "" ? def.clawback : undefined;
-    validateSingle(accounts, filename, def);
-  }
-}
-
-export function validateASADefs (
-  obj: Object, accounts: AccountMap | RuntimeAccountMap, filename: string): ASADefs {
+/**
+ * Validate and parse each field of asset definition
+ * @param asaDef asset definition
+ * @param source source of assetDef: asa.yaml file OR function deployASA
+ * @returns parsed asa definition
+ */
+function _parseASADef (asaDef: ASADef, source?: string): ASADef {
   try {
-    const parsed = ASADefsSchema.parse(obj);
-    Object.keys(parsed).forEach(k => {
-      if (parsed[k].defaultFrozen === undefined) {
-        parsed[k].defaultFrozen = false;
-      }
-    });
-    validateParsedASADefs(accounts, parsed, filename);
-    return parsed;
+    const parsedDef = ASADefSchema.parse(asaDef);
+    parsedDef.manager = parsedDef.manager !== "" ? parsedDef.manager : undefined;
+    parsedDef.reserve = parsedDef.reserve !== "" ? parsedDef.reserve : undefined;
+    parsedDef.freeze = parsedDef.freeze !== "" ? parsedDef.freeze : undefined;
+    parsedDef.clawback = parsedDef.clawback !== "" ? parsedDef.clawback : undefined;
+    parsedDef.defaultFrozen = parsedDef.defaultFrozen ?? false;
+    return parsedDef;
   } catch (e) {
     if (e instanceof z.ZodError) {
       throw new RuntimeError(
         RUNTIME_ERRORS.ASA.PARAM_PARSE_ERROR, {
           reason: parseZodError(e),
-          filename: filename
+          source: source
         }, e);
     }
     throw e;
   }
 }
 
+/**
+ * Override & validate ASA definition in asa.yaml using custom params passed via deployASA
+ * @param accounts accounts by name
+ * @param sourceDef source asset definition (in asa.yaml)
+ * @param customDef custom asset def params (passed during ASA deployment)
```

Comment:
sourceDef and customDef  doesn't sound very clear. How about renaming to: `origDef` and `newDef`? Or maybe something else?

Reply:
`origDef` & `newDef` sounds more clear.